druid/src/widget/slider.rs

```diff
@@ -78,29 +127,477 @@ impl Slider {
+    fn layout(&mut self, ctx: &mut LayoutCtx, bc: &BoxConstraints, _data: &f64, env: &Env) -> Size {
+        bc.debug_check("Slider");
+        slider_layout(ctx, bc, env, self.mapping)
+    }
+
+    #[instrument(name = "Slider", level = "trace", skip(self, ctx, data, env))]
+    fn paint(&mut self, ctx: &mut PaintCtx, data: &f64, env: &Env) {
+        paint_slider_background(ctx, 0.0, *data, &self.track_color, self.mapping, env);
+
+        self.knob
+            .paint(ctx, *data, env, self.mapping, self.knob_style);
+    }
+
+    fn debug_state(&self, data: &f64) -> DebugState {
+        DebugState {
+            display_name: self.short_type_name().to_string(),
+            main_value: data.to_string(),
+            ..Default::default()
+        }
+    }
+}
+
+impl RangeSlider {
+    /// Create a new `RangeSlider`.
+    pub fn new() -> RangeSlider {
+        Default::default()
+    }
+
+    /// Builder-style method to set the range covered by this range slider.
+    ///
+    /// The default range is `0.0..1.0`.
+    pub fn with_range(mut self, min: f64, max: f64) -> Self {
+        self.mapping.min = min;
+        self.mapping.max = max;
+        self
+    }
+
+    /// Builder-style method to set the stepping.
+    ///
+    /// The default step size is `0.0` (smooth).
+    pub fn with_step(mut self, step: f64) -> Self {
+        if step < 0.0 {
+            warn!("bad stepping (must be positive): {}", step);
+            return self;
+        }
+        self.mapping.step = if step > 0.0 {
+            Some(step)
+        } else {
+            // A stepping value of 0.0 would yield an infinite amount of steps.
+            // Enforce no stepping instead.
+            None
+        };
+        self
+    }
+
+    /// Builder-style method to set the track color.
+    ///
+    /// The default color is `None`.
+    pub fn track_color(mut self, color: impl Into<Option<KeyOrValue<Color>>>) -> Self {
+        self.track_color = color.into();
+        self
+    }
+
+    /// Builder-style method to set the knob style.
+    ///
+    /// The default is `Circle`.
+    pub fn knob_style(mut self, knob_style: KnobStyle) -> Self {
+        self.knob_style = knob_style;
+        self
+    }
+
+    /// Builder-style method to the the axis on which the slider moves.
+    ///
+    /// The default is `Horizontal`.
+    pub fn axis(mut self, axis: Axis) -> Self {
+        self.mapping.axis = axis;
+        self
+    }
+
+    /// Returns the Mapping of this Slider.
+    pub fn get_mapping(&self) -> SliderValueMapping {
+        self.mapping
+    }
+
+    /// Builder-style method to create an annotated range slider.
+    ///
+    pub fn annotated(self, named_steps: f64, unnamed_steps: f64) -> Annotated<(f64, f64), Self> {
+        let mapping = self.mapping;
+        Annotated::new(self, mapping, named_steps, unnamed_steps)
+    }
+}
+
+impl Widget<(f64, f64)> for RangeSlider {
+    #[instrument(name = "Slider", level = "trace", skip(self, ctx, event, data, env))]
```

Comment:
```suggestion
    #[instrument(name = "RangeSlider", level = "trace", skip(self, ctx, event, data, env))]
```

(Note that I'm starting to regret adding `instrument` annotations to druid, precisely because of this kind of overhead, but as long as they're here they should remain consistent)